### PR TITLE
Add browser compat data for linear-gradient()

### PR DESCRIPTION
**Pull request by Elchi3**
Opening on behalf of @teoli2003

### CODE REVIEW

File: css/types/image.json [hunk 1]
```diff
@@ -105,6 +105,293 @@
...
+                  },
+                  "edge": {
+                    "version_added": "12"
+                  },
+                  "edge_mobile": {
+                    "version_added": "12"
+                  },
+                  "firefox": {
+                    "version_added": "55",
+                    "notes": "Unitless <code>0</code> angles are supported since Firefox 46 in prefixed versions only."
```
Comment (ExE-Boss) [inline css/types/image.json]: I think a better wording would be:
```json
"firefox": [
  {
    "version_added": "55"
  },
  {
    "version_added": "46",
    "partial_implementation": true,
    "notes": "Supported only in <code>-moz-linear-gradient(…)</code>"
  }
]
```

File: css/types/image.json [hunk 2]
```diff
@@ -105,6 +105,293 @@
...
+                        "type": "preference",
+                        "name": "layout.css.prefixes.webkit",
+                        "value_to_set": "true"
+                      }
+                    ]
+                  }
+                ],
+                "ie": {
+                  "version_added": "10",
+                  "notes": "Internet Explorer 5.5 through 9.0 supported gradients via a proprietary filter: <code><a href='http://msdn.microsoft.com/en-us/library/ms532997(VS.85,loband).aspx'>filter: progid:DXImageTransform.Microsoft.Gradient()</a></code>."
```
Comment (ExE-Boss) [inline css/types/image.json]: I think this `<a>` should be changed to point to: https://developer.mozilla.org/docs/Web/CSS/-ms-filter#Gradient

File: css/types/image.json (hunk 1 above)
Comment (Jean-Yves-P) [reply css/types/image.json]: That makes sense, fixed.

File: css/types/image.json (hunk 2 above)
Comment (Jean-Yves-P) [reply css/types/image.json]: Good catch. The original link was added before we created the relevant MDN page. Fixed.

File: css/types/image.json [hunk 3]
```diff
@@ -105,6 +105,306 @@
...
+                  "deprecated": false
+                }
+              }
+            },
+            "interpolation_hints": {
+              "__compat": {
+                "description": "Interpolation Hints / Gradient Midpoints",
+                "support": {
+                  "webview_android": {
+                    "version_added": true
```
Comment (jpmedley) [inline css/types/image.json]: This and chrome_android should also be 40.

Comment (teoli2003) [reply css/types/image.json]: Done

**Pull request merged**